openwisp_monitoring/check/tests/test_iperf.py

```diff
@@ -620,9 +599,115 @@ def test_iperf_check_auth_config(
                 mock_warn.assert_has_calls(self._EXPECTED_WARN_CALLS)
                 mock_exec_command.assert_has_calls(self._EXPECTED_COMMAND_CALLS)
-                self.assertEqual(mock_get_iperf_servers.call_count, 1)
+
+    @patch.object(Ssh, 'exec_command')
+    @patch.object(iperf_logger, 'warning')
+    @patch.object(cache, 'set')
+    @patch.object(task_logger, 'warning')
+    def test_iperf_check_task_with_multiple_server_config(self, *args):
+        check, _ = self._create_iperf_test_env()
+        org = self.device.organization
+        mock_warn = args[2]
+        mock_set = args[1]
+        mock_task_logger = args[0]
+        mock_exec_command = args[3]
+        iperf_multiple_server_config = {
+            str(org.id): {'host': self._IPERF_TEST_MULTIPLE_SERVERS}
+        }
+        self._EXPECTED_COMMAND_CALLS_SERVER_1 = [
+            call(
+                f'iperf3 -c {self._IPERF_TEST_MULTIPLE_SERVERS[0]} -p 5201 -t 10 -b 0 -J',
+                raise_unexpected_exit=False,
+            ),
+            call(
+                f'iperf3 -c {self._IPERF_TEST_MULTIPLE_SERVERS[0]} -p 5201 -t 10 -b 30M -u -J',
+                raise_unexpected_exit=False,
+            ),
+        ]
+        self._EXPECTED_COMMAND_CALLS_SERVER_2 = [
+            call(
+                f'iperf3 -c {self._IPERF_TEST_MULTIPLE_SERVERS[1]} -p 5201 -t 10 -b 0 -J',
+                raise_unexpected_exit=False,
+            ),
+            call(
+                f'iperf3 -c {self._IPERF_TEST_MULTIPLE_SERVERS[1]} -p 5201 -t 10 -b 30M -u -J',
+                raise_unexpected_exit=False,
+            ),
+        ]
+
+        with self.subTest('Test iperf check without config'):
+            management.call_command('run_checks')
+            mock_task_logger.assert_called_with(
+                (
+                    f'Iperf servers for organization "{org}"'
+                    f'is not configured properly, iperf check skipped!'
+                )
+            )
+            mock_task_logger.reset_mock()
+
+        with patch.object(
+            app_settings, 'IPERF_CHECK_CONFIG', iperf_multiple_server_config
+        ):
+            with self.subTest('Test iperf check when all iperf servers are available'):
+                mock_set.return_value = True
+                mock_exec_command.side_effect = [(RESULT_TCP, 0), (RESULT_UDP, 0)]
+                management.call_command('run_checks')
+                self.assertEqual(mock_warn.call_count, 0)
+                self.assertEqual(mock_task_logger.call_count, 0)
+                self.assertEqual(mock_set.call_count, 1)
+                self.assertEqual(mock_exec_command.call_count, 2)
+                mock_exec_command.assert_has_calls(
+                    self._EXPECTED_COMMAND_CALLS_SERVER_1
+                )
+                mock_set.reset_mock()
+                mock_warn.reset_mock()
+                mock_task_logger.reset_mock()
+                mock_exec_command.reset_mock()
```

Comment:
Always do resetting of test outside of the `subTest`. In the current code, if this subTest fails for some reason, then the mocks would not be reset which would make the following sub-tests to fail as well. 

```suggestion
            mock_set.reset_mock()
            mock_warn.reset_mock()
            mock_task_logger.reset_mock()
            mock_exec_command.reset_mock()
```

Reply:
I'll update it :+1: